lib/range.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+const rangeParser = require('parse-numeric-range');
+const union = require('array-union');
+
+function parseRange(files) {
+	const any = new Set();
+	const ranges = {};
+	const filenames = [];
+
+	for (const file of files) {
+		const [actualFilename, rangeStr] = splitRangeStr(file);
+
+		if (any.has(actualFilename)) {
+			continue;
+		}
+
+		if (rangeStr === '') {
+			any.add(actualFilename);
+			ranges[actualFilename] = [];
+			continue;
+		}
+
+		const lines = rangeParser.parse(rangeStr);
+
+		if (ranges[actualFilename]) {
+			ranges[actualFilename] = union(ranges[actualFilename], lines);
+		} else {
+			ranges[actualFilename] = lines;
+		}
+
+		filenames.push(actualFilename);
+	}
+
+	return {
+		filenames: union(filenames, ...any),
+		ranges
+	};
+}
+
+exports.parseRange = parseRange;
+
+const rangeRegExp = /^(\d+(-\d+)?,?)+$/;
+const splitRangeStrCache = new Map();
+
+function splitRangeStr(file) {
+	if (splitRangeStrCache.has(file)) {
+		return [...splitRangeStrCache.get(file)];
+	}
+
+	const colonIndex = file.lastIndexOf(':');
+	const mightHaveRange = colonIndex > -1 && file[colonIndex + 1] !== '\\';
```

Comment:
That regex I'm using might be a bit heavy so I use a fast and dirty check first and store the result in `mightHaveRange`, also there is no need for a test if you remove this check (and the regex), most of the tests will fail on windows.

Reply:
I'm not too fussed about performance overheads, presumably you only run this with a handful of ranges so it'll be really hard to tell what's faster or slower.

> here is no need for a test if you remove this check (and the regex), most of the tests will fail on windows.

Are you saying the test is not necessary, or it is because of Windows?

Reply:
you can look at this from when this check was not there https://travis-ci.org/avajs/ava/builds/556309917

Reply:
Right, so let's add a comment explaining this handles file paths on Windows that do not include ranges.